### PR TITLE
add cpp solution for 989

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-989.mdx [hunk 1]
```diff
@@ -51,4 +51,51 @@ class Main {
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("race.in", "r", stdin);
+	freopen("race.out", "w", stdout);
+	
+	int k, n, x, ans, res, i;
```
Comment (SansPapyrus683) [inline solutions/usaco-989.mdx:64]: for the love of god please don't put all your variables in the same place

File: solutions/usaco-989.mdx [hunk 2]
```diff
@@ -51,4 +51,51 @@ class Main {
...
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("race.in", "r", stdin);
+	freopen("race.out", "w", stdout);
```
Comment (SansPapyrus683) [inline solutions/usaco-989.mdx:62]: this is optional, but i'd like it if you could use `std::ifstream` and `std::ofstream`
completely up to you, it wouldn't be fair to mandate that

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-989.mdx (hunk 1 above)
Comment (jjliewie) [reply solutions/usaco-989.mdx:64]: I'll change it.

File: solutions/usaco-989.mdx [hunk 3]
```diff
@@ -51,4 +51,54 @@ class Main {
...
+	
+		int i = 1; 
+		while (i < x && res < k) {
+			res += i;
+			ans++;
+			i++;
+		}
+	
+		i = x;
+
```
Comment (ryanchou-dev) [inline solutions/usaco-989.mdx:83]:
```suggestion
```

**Review by ryanchou-dev — APPROVED** [review]
⭐

File: solutions/usaco-989.mdx [hunk 4]
```diff
@@ -51,4 +51,53 @@ class Main {
...
+	
+		int i = 1; 
+		while (i < x && res < k) {
+			res += i;
+			ans++;
+			i++;
+		}
+	
+		i = x;
+		while (res < k) {
```
Comment (SansPapyrus683) [inline solutions/usaco-989.mdx:81]: comments

Comment (jjliewie) [reply solutions/usaco-989.mdx:82]: is it better now

**Comment by SansPapyrus683** [comment]
oof uh i actually already made a solution for this, sorry!

**Pull request closed without merging**